zerver/lib/actions.py

```diff
@@ -6420,6 +6420,14 @@ def do_mark_hotspot_as_read(user: UserProfile, hotspot: str) -> None:
     send_event(user.realm, event, [user.id])
 
 
+def do_reset_all_hotspots(user: UserProfile) -> None:
+    user.tutorial_status = UserProfile.TUTORIAL_STARTED
+    user.save(update_fields=["tutorial_status"])
+    UserHotspot.objects.filter(user=user).delete()
+    event = dict(type="hotspots", hotspots=get_next_hotspots(user))
+    send_event(user.realm, event, [user.id])
```

Comment:
So, one thing we should think about here is whether the right implementation is to delete all UserHotspot objects, or something more tactical.  

I could imagine us implementing new hotspots in the future that aren't strictly the initial tutorial (E.g. to advertise new features), and I wonder if we'd be better off filtering to `__in=list(ALL_HOTSPOTS.keys())` here, maybe with a prep commit to rename ALL_HOTSPOTS=>INTRO_HOTSPOTS.

Reply:
@timabbott  A couple of follow-up questions here:
1. Would it be better if we extracted INTRO_HOTSPOTS from ALL_HOTSPOTS, but keeping the former dict as an aggregate for all types of hotspots we may add in the future.
```
INTRO_HOTSPOTS: Dict[str, Dict[str, Promise]] = {
      "intro_reply": {...},
       ...
}

ALL_HOTSPOTS: Dict[str, Dict[str, Promise]] = {
     **INTRO_HOTSPOTS,
}
```
2. Should the way we handle `ALWAYS_SEND_ALL_HOTSPOTS=True` change? I suppose implementing it in the way described in the previous point would allow us to keep it the same.
3. For the delete, you mentioned filtering on ALL_HOTSPOTS key above. Did you mean filtering on `__in=list(INTRO_HOTSPOTS.keys())` instead?

Reply:
1 -- yes, that's probably a good idea.
2 -- `ALWAYS_SEND_ALL_HOTSPOTS` is just for development, so I'd do whatever makes that convenient.
3 -- Yeah, I meant the `INTRO_HOTSPOTS` list, at least if we're doing 1.